source/common/network/dns_impl.cc

```diff
@@ -168,6 +169,7 @@ void DnsResolverImpl::PendingResolution::onAresGetAddrInfoCallback(int status, i
 
   if (completed_) {
     if (!cancelled_) {
+      // TODO(chaoqin-li1123): remove this exception catching since the process is dying anyway.
```

Comment:
This comment is not accurate. This can happen during normal operation for some of these I think. Doesn't this need some AUDIT statement or CI should fail? Same below?

Reply:
Thanks, will update it.

Reply:
This file is in `RAW_TRY_ALLOWLIST` in check_format.py. However I'm not sure if this is justified and think maybe it should be a NEEDS_AUDIT.

Do we need to allow the callback to throw? Or can we just have the callbacks return an error?

I assume it isn't c_ares that forces us to handle exceptions here?

Reply:
This code is executed in main thread and in dns filter, I think it is fine to allow a filter to throw and catch.

Reply:
Thanks, comment updated.